src/vonage/application.py

```diff
@@ -0,0 +1,166 @@
+import logging
+
+from requests.adapters import HTTPAdapter
+from requests.sessions import Session
+
+from .errors import AuthenticationError, ClientError, ServerError
+
+from deprecated import deprecated
+
+try:
+    from json import JSONDecodeError
+except ImportError:
+    JSONDecodeError = ValueError
+
+logger = logging.getLogger("vonage")
+
+
+class BasicAuthenticatedServer(object):
```

Comment:
Actually this seems a potentially "better" way of doing it; Let the client rely on the API class, not vice-versa